File: Models/DTOs/RatingDTO.cs
using GoWheels_WebAPI.Models.Entities;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System.ComponentModel.DataAnnotations;

namespace GoWheels_WebAPI.Models.DTOs
{
    public class RatingDTO
    {
        [Required(ErrorMessage = "Comment is required")]
        public string? Comment { get; set; }

        [Required(ErrorMessage = "Point is required")]
        [Range(1, 5, ErrorMessage = "Point must be between 1 and 5")]
        public float Point { get; set; }

        [Required(ErrorMessage = "PostId is required")]
        public int PostId { get; set; }
        public required string UserId { get; set; }
    }
}

File: FE/src/DTOs/RatingDto.js
// src/dtos/RatingDTO.js

export default class RatingDTO {
  constructor(data = {}) {
    this.id = data.id || 0;
      this.comment = data.comment || '';
      this.point = data.point || 1;
      this.postId = data.postId || 0;
  }

  toFormData() {
      const formData = new FormData();
      if (this.id !== 0) {  // Kiểm tra xem `id` có khác null không
        formData.append("Id", this.id);
    }
      formData.append("Comment", this.comment);
      formData.append("Point", this.point);
      formData.append("PostId", this.postId);
      return formData;
  }
}
